subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultLenientConfiguration.java

```diff
@@ -335,6 +326,7 @@ public void visitFile(ComponentArtifactIdentifier artifactIdentifier, AttributeC
         }
 
         public Set<File> getFiles() {
+            addArtifactsFiles();
```

Comment:
I thought about that, but since we use a set it should be ok, right?

Anyway, I did what you proposed as it is also in line with the implementation in `ResolvedFilesCollectingVisitor`. Lets keep that consistent.